source/Calamari/ArgoCD/Git/RepositoryWrapper.cs

```diff
@@ -91,23 +90,63 @@ public async Task PushChanges(bool requiresPullRequest,
                                       CancellationToken cancellationToken)
         {
             var currentBranchName = repository.GetBranchName(branchName);
-            var pushToBranchName = currentBranchName;
-            if (requiresPullRequest)
-            {
-                pushToBranchName = new GitBranchName(CalculateBranchName());
-            }
+            var commit = repository.Head.Tip; // We should have just pushed to the tip of this branch
+
+            var pushToBranchName = requiresPullRequest ? 
```

Comment:
if we are required to create a PR - but can't create a vendorApi - should we throw rather than pushing branches to their remote?
This could (potentially) be detected at repository creation.